src/data/equiv/basic.lean

```diff
@@ -55,6 +55,9 @@ coe_fn_injective (funext H)
 @[ext] lemma perm.ext {σ τ : equiv.perm α} (H : ∀ x, σ x = τ x) : σ = τ :=
 equiv.ext H
 
+lemma perm.ext_iff {σ τ :equiv.perm α} : σ = τ ↔ ∀ x, σ x = τ x :=
+by {split, finish, exact perm.ext}
+
```

Comment:
A few mathlib style comments: make sure to have a space after `:`, after `by {`, and before the final `}`.

You might also define `ext_iff` for `equiv` itself while you're at it.  In the fine art of code golf, here are some term-mode proofs for both:

```lean
lemma ext_iff {f g : equiv α β} : f = g ↔ ∀ x, f x = g x :=
⟨λ h x, h ▸ rfl, ext⟩

lemma perm.ext_iff {σ τ : equiv.perm α} : σ = τ ↔ ∀ x, σ x = τ x :=
⟨λ h x, h ▸ rfl, perm.ext⟩
```